community/ssl/src/test/java/org/neo4j/ssl/SslResourceBuilder.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.ssl;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.neo4j.io.fs.DefaultFileSystemAbstraction;
+import org.neo4j.io.fs.FileSystemAbstraction;
+
+import static org.neo4j.ssl.SslResourceBuilder.SignedBy.CA;
+import static org.neo4j.ssl.SslResourceBuilder.SignedBy.ROOT;
+import static org.neo4j.ssl.SslResourceBuilder.SignedBy.SELF;
+
+/**
+ * This builder has a finite set of pre-generated resource
+ * keys and certificates which can be utilized in tests.
+ *
+ * The script gen-root.sh is used to generate the root
+ * resources and gey-key.sh is used to generate the
+ * instance resources.
+ */
+public class SslResourceBuilder
+{
+    private static final String ROOT_CERTIFICATE_NAME = "root.crt";
```

Comment:
Can we create these keys and certs in this class, rather than commit the key and cert files into the source code?

The reason for asking for this is 1) It is hard to understand what is in the committed key and cert files. 2) If any tests failed due to errors in pre-created cert/key such as cert/key expiration, then it is hard to debug tests.